.github/workflows/testing.yml

```diff
@@ -1,41 +1,21 @@
-name: "Testing Suite"
+name: "Testing"
 on: [push, pull_request]
 
 jobs:
-  unit-test:
-    permissions:
-      contents: write
+  test:
     strategy:
       matrix:
-        platform: [macos-latest, ubuntu-20.04, windows-latest]
-
-    runs-on: ${{ matrix.platform }}
+        os: [macos-latest, ubuntu-latest, windows-latest]
+    runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@v4
         with:
           submodules: true
-
-      - name: Initialize Rust Cache
```

Comment:
Could you keep the cache? Windows job takes 3 minutes, I suspect this will only grow with time.

Reply:
Good point, the time has grown a little because jobs are now running sequentially. Since you've noticed that, I've re-added the cache but with properly set-up Rust action. Now cache is working and the times are noticeably better!

Reply:
Also I've noticed the time tends to fluctuate a bit, depending on Github load and whatnot.